src/main/python/fine_tuning/run_batch.py

```diff
@@ -0,0 +1,211 @@
+# -*- coding: utf-8 -*-
+"""
+Anserini: A toolkit for reproducible information retrieval research built on Lucene
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import os
+import subprocess
+import argparse
+from multiprocessing import Pool
+import json
+import logging
+
+import yaml
+
+from search import Search
+from evaluation import Evaluation
+from effectiveness import Effectiveness
+
+logger = logging.getLogger('fine_tuning')
+logger.setLevel(logging.INFO)
+# create console handler with a higher log level
+ch = logging.StreamHandler()
+ch.setLevel(logging.INFO)
+formatter = logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s')
+ch.setFormatter(formatter)
+# add the handlers to the logger
+logger.addHandler(ch)
+
+parallelism=1
+def batch_everything(all_params, func):
+    if len(all_params) == 0:
+        return
+    p = Pool(parallelism)
+    p.map(func, all_params)
+
+def isclose(a, b, rel_tol=1e-09, abs_tol=0.0):
+    return abs(a-b) <= max(rel_tol * max(abs(a), abs(b)), abs_tol)
+
+def get_index_path(yaml_data):
+    """
+    Find the possible index path
+    """
+    index_path = os.path.join('lucene-index.{0}.pos+docvectors{1}'.format(yaml_data['name'], \
+                                                                          '+rawdocs' if '-storeRawDocs' in yaml_data['index_options'] else ''))
+    if not os.path.exists(index_path):
+        index_path = yaml_data['index_path']
+        if not index_path or not os.path.exists(index_path):
+            for input_root in yaml_data['input_roots']:
+                if os.path.exists(os.path.join(input_root, yaml_data['index_path'])):
+                    index_path = os.path.join(input_root, yaml_data['index_path'])
+                    break
+    return index_path
+
+def batch_retrieval(collection_yaml, models_yaml, output_root):
+    all_params = []
+    program = os.path.join(collection_yaml['anserini_root'], 'target/appassembler/bin', 'SearchCollection')
+    index_path = get_index_path(collection_yaml)
+    this_output_root = os.path.join(output_root, collection_yaml['name'])
```

Comment:
This part seems to be redundant in all the `batch_` functions. Moving it out to main?

Reply:
okay